controllers/CommentOnTextController.js

```diff
@@ -0,0 +1,118 @@
+import { Comment, User, Article } from '../database/models';
+import newInteractionNotification from '../helpers/notification/newInteractionNotification';
+
+/**
+ * @description Comment on highlighted text from article Controller class
+ */
+class CommentOnTextController {
+  /**
+   * @author Daniel and Manzi
+   * @param {*} req
+   * @param {*} res
+   * @returns {Object} object
+   */
+  static async addComment(req, res) {
+    const { articleSlug } = req.params;
+    const {
+      highlightedText,
+      body, endPoint,
+      startPoint
+    } = req.body.comment;
+    const { currentUser } = req;
+
+    const article = await Article.findOne({
+      where: {
+        slug: articleSlug
```

Comment:
@Nziranziza To avoid commenting on an unpublished article, please check the status of an article in your query! `status: 'published' `

Reply:
@mcaleb808 thank you! that is true